include/appbase/execution_priority_queue.hpp

```diff
@@ -100,13 +143,14 @@ class execution_priority_queue : public boost::asio::execution_context
    private:
       execution_priority_queue& context_;
       int priority_;
+      int index_;
    };
 
    template <typename Function>
    boost::asio::executor_binder<Function, executor>
-   wrap(int priority, Function&& func)
+   wrap(int priority, Function&& func, int index = 0)
```

Comment:
I find `index` an odd interface here. I think this is too generic of an interface. In my mind making the two queues a first class feature instead of this generic approach would result in cleaner more optimal code.

Reply:
Could you elaborate this? Thanks.

Reply:
I like Greg's suggestion on a different interface. In general a think a less generic interface would be better. Currently passing in an out of range index would cause a crash (but we don't want to pay for checking valid index). I think just add two queue support and make that the only option provided. Why bother with providing a single priority queue option. A non-nodeos user will just always have an empty second queue.